extensions/sld/src/ConnectablePositionAdder.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/sld/ConnectablePositionAdder.hpp>
+
+#include <powsybl/iidm/Connectable.hpp>
+#include <powsybl/iidm/extensions/sld/ConnectablePosition.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace sld {
+
+ConnectablePositionAdder::ConnectablePositionAdder(Extendable& extendable) :
+    ExtensionAdder(extendable) {
+}
+
+ConnectablePositionAdder::FeederAdder::FeederAdder(ConnectablePositionAdder& parent, ConnectablePosition::OptionalFeeder& feeder) :
+    m_parent(parent),
+    m_feeder(feeder) {
+}
+
+ConnectablePositionAdder& ConnectablePositionAdder::FeederAdder::add() {
+    if (!m_order.is_initialized()) {
+        throw PowsyblException("Feeder order is not set");
+    }
+    if (!m_direction.is_initialized()) {
+        throw PowsyblException("Feeder direction is not set");
+    }
+    m_feeder.get() = ConnectablePosition::Feeder(m_name, *m_order, *m_direction);
+    return m_parent.get();
```

Comment:
Does it work?
```
m_feeder = ConnectablePosition::Feeder(m_name, *m_order, *m_direction);
```

I think you should also check if the name is not empty. Looking to the java code, we only check if the name is null, but an empty string is weird

Reply:
>Does it work?
To be checked. It looks like this code should be written (as suggested [here](https://github.com/powsybl/powsybl-iidm4cpp/pull/215#discussion_r525250627) to respect Java implementation to avoid memory corruption...

> I think you should also check if the name is not empty. Looking to the java code, we only check if the name is null, but an empty string is weird
I agree, I wanted to be stricly Java compliant. => check added